internal/reconciler/logpipeline/reconciler.go

```diff
@@ -163,6 +167,17 @@ func (r *Reconciler) doReconcile(ctx context.Context, pipeline *telemetryv1alpha
 	}
 
 	reconcilablePipelines := r.getReconcilablePipelines(ctx, allPipelines.Items)
+	if len(reconcilablePipelines) == 0 {
+		logf.FromContext(ctx).V(1).Info("cleaning up log pipeline resources: all log pipelines are non-reconcilable")
+		if err = r.deleteResources(ctx); err != nil {
```

Comment:
Maybe we should not return any "not found" errors to the controller-runtime. This might cause continuous re-queueing.

Reply:
not found errors are ignored